lib/rss.js

```diff
@@ -0,0 +1,192 @@
+const RSS = require('rss')
+const generateDiff = require('deep-diff').diff
+const fs = require('fs')
+
+function createLeadersHash(array) {
+  return array.reduce((hash, elem) => {
+    hash[elem.id] = elem
+    return hash
+  }, {})
+}
+
+function createOrgHash(array) {
+  let rank = 0
+  return array.reduce((hash, elem) => {
+    rank++
+    elem.rank = rank
+    elem.leaders = createLeadersHash(elem.leaders)
+    hash[elem.slug] = elem
+    return hash
+  }, {})
+}
+
+let current = {}
+try {
+  current = JSON.parse(
+    fs.readFileSync(`${__dirname}/../out/feed_items.json`).toString()
+  )
+} catch (e) {
+  current = {}
+}
+
+const buildInfo = JSON.parse(
+  fs.readFileSync(`${__dirname}/../out/scrape_info.json`).toString()
+)
+
+const oldData = JSON.parse(
+  fs.readFileSync(`${__dirname}/../out/data_old.json`).toString()
+)
+const newData = JSON.parse(
+  fs.readFileSync(`${__dirname}/../out/data.json`).toString()
```

Comment:
Every new .js is loading this file again.  And creating a new script , and a longer `build` command.

If these .js are modularised, the data can be parsed once and passed around .

Reply:
Should I modularize in this PR? Seems like a very big change. After this gets merged in, I could definitely do that in a new PR though so as not to clutter this one with too many changes.

Reply:
yes, you should modularise the new code in this PR.

Reply:
And we have tests now.  The bar has been raised :P